icu4c/source/test/intltest/units_test.cpp

```diff
@@ -247,9 +247,12 @@ void UnitsTest::testConverter() {
             continue;
         }
 
+        double maxDelta = 1e-6 * uprv_fabs(testCase.expectedValue);
```

Comment:
We need to stick with specific level of accuracy. For example, if we decided that the deviation  could be 1e-4, then we should make the test cases follow that.

Reply:
1e-4 is a terrible accuracy. I was already disappointed that this seems to have a delta bigger than 1e-7, but I'm indeed not digging into it yet, not considering it a part of this PR.

I did take this opportunity to show how accurate our test currently are, so we could also catch regressions when anything gets worse than this.